internal/gl/gl_js.go

```diff
@@ -351,6 +357,9 @@ func (f *Functions) LinkProgram(p Program) {
 	f.Ctx.Call("linkProgram", js.Value(p))
 }
 func (f *Functions) PixelStorei(pname Enum, param int) {
+	if !f.isWebGL2 {
+		return // PACK_ROW_LENGTH and UNPACK_ROW_LENGTH is only available on WebGL 2
```

Comment:
You're still ignoring PixelStorei which may corrupt texture downloads and uploads. How about skipping the call to PixelStore if row length == image width? Or, replace value with 0 if so.